WSLBlobNFS/WSLBlobNFS.psm1

```diff
@@ -6,178 +11,539 @@ Set-PSDebug -Strict
+# - Sign the module
+#   https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_output_streams?view=powershell-5.1
+# - Help content for each function: Get-Help <function-name>
+# - Add Uninitialize module support
 
-function Install-WSLBlobNFS
+# To-do:
+# - To handle -Debug issue, set the debug preference to continue and then set it back to the original value.
+# - Check why -Debug switch is not working as expected in testing env:
+#  https://stackoverflow.com/questions/4301562/how-to-properly-use-the-verbose-and-debug-parameters-in-a-custom-cmdlet
+
+
+#
+# Internal functions
+#
+function Get-ModuleVersion
 {
+    [CmdletBinding()]
+    param()
+
+    # In local dev env, the module is not installed from gallery. Hence, Get-InstalledModule will fail.
+    # Hence, suppress the error and return 0.0.0 as the module version.
+    $blobNFSModule = Get-InstalledModule -Name $moduleName 2>$null
+    if ($blobNFSModule -eq $null)
+    {
+        return "0.0.0"
+    }
+
+    return $blobNFSModule.Version.ToString()
+}
+
+function Write-Success
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)][string]$message
+    )
+
+    Write-Host $message -ForegroundColor DarkGreen
+}
+
+function Execute-WSL
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)][string]$wslcommand
+    )
+
+    Write-Debug "Executing $wslcommand"
+    wsl -d $distroName -u $userName -e bash -c $wslcommand
+}
+
+function Copy-WSLFiles
+{
+    [CmdletBinding()]
+    param()
+
+    # Note: Quote the path with '' to preserve space
+    Write-Verbose "Copying necessary linux files from $modulePathForLinux"
+
+    Execute-WSL "mkdir -p $linuxScriptsPath"
+
+    Execute-WSL "cp '$modulePathForLinux/$wslScriptName' $linuxScriptsPath/$wslScriptName"
+
+    # Files saved from windows will have \r\n line endings. Hence, we need to remove \r.
+    Execute-WSL "sed -i -e 's/\r$//' $linuxScriptsPath/$wslScriptName"
+
+    Execute-WSL "chmod +x $linuxScriptsPath/$wslScriptName"
+
+    Execute-WSL "cp '$modulePathForLinux/$queryScriptName' $linuxScriptsPath/$queryScriptName"
+
+    # Files saved from windows will have \r\n line endings. Hence, we need to remove \r.
+    Execute-WSL "sed -i -e 's/\r$//' $linuxScriptsPath/$queryScriptName"
+
+    Execute-WSL "chmod +x $linuxScriptsPath/$queryScriptName"
+
+    Execute-WSL "ls -l $linuxScriptsPath | grep -E '$wslScriptName|$queryScriptName' > /dev/null 2>&1"
+}
+
+function Intall-WSLBlobNFS-Internal
+{
+    [CmdletBinding()]
+    param()
+
     # Check if wsl is installed or not.
-    $wslstatus = wsl -l -v
-    if($LASTEXITCODE -eq 1)
+    wsl -l -v > $null 2>&1
+    $wslstatus = $LASTEXITCODE
+
+    # Check if the distro is installed or not.
+    # Note: executing a command such as ls is required else if the distro is installed, then control will not come out of wsl command.
+    wsl -d $distroName ls > $null 2>&1
+    $distrostatus = $LASTEXITCODE
```

Comment:
if wsl is not installed, is it safe to run this command, or shall we run this only if wslStatus is success

Reply:
it's safe to run, but anwyay i've changed the logic to do such checks one after the other just to be safe.